api/v1alpha1/project_config_types.go

```diff
@@ -120,6 +127,20 @@ type GitHubWebhookReceiverConfig struct {
 	SecretRef corev1.LocalObjectReference `json:"secretRef" protobuf:"bytes,1,opt,name=secretRef"`
 }
 
+// QuayWebhookReceiverConfig describes a webhook receiver that is compatible
+// with Quay.io payloads.
+type QuayWebhookReceiverConfig struct {
+	// SecretRef contains a reference to a Secret. For Project-scoped webhook
+	// receivers, the referenced Secret must be in the same namespace as the
+	// ProjectConfig.
+	//
+	// For cluster-scoped webhook receivers, the referenced Secret must be in the
+	// designated "cluster Secrets" namespace.
+	//
+	// The Secret's data map is expected to contain a `secret` key.
```

Comment:
Comments on these types eventually wind up in some generated CRD docs...

More info about what this is supposed to match would be helpful. For GitHub, we're saying this:

```
	// The Secret's data map is expected to contain a `secret` key whose value is
	// the shared secret used to authenticate the webhook requests sent by GitHub.
	// For more information please refer to GitHub documentation:
	//   https://docs.github.com/en/webhooks/using-webhooks/validating-webhook-deliveries
```

```suggestion
	// The Secret's data map is expected to contain a `secret` key.
```

Reply:
I've updated it to reflect the above.

Reply:
Ok. Similar situation to Docker Hub, huh?

This is what #4351 says. Perhaps just adapt this.

```
	// The Secret's data map is expected to contain a `secret` key whose value
	// does NOT need to be shared directly with Docker Hub when registering a
	// webhook. It is used only by Kargo to create a complex, hard-to-guess URL,
	// which implicitly serves as a shared secret. For more information about
	// Docker Hub webhooks, please refer to the Docker documentation:
	//   https://docs.docker.com/docker-hub/webhooks/
```